src/cgutils.cpp

```diff
@@ -2519,7 +2521,8 @@ static Value *emit_arrayptr_internal(jl_codectx_t &ctx, const jl_cgval_t &tinfo,
                 PointerType::get(PPT->getElementType(), AS),
                 PT->getAddressSpace()));
     }
-    LoadInst *LI = ctx.builder.CreateAlignedLoad(addr, Align(sizeof(char*)));
+    LoadInst *LI = ctx.builder.CreateAlignedLoad(
+        cast<PointerType>(addr->getType())->getElementType(), addr, Align(sizeof(char*)));
```

Comment:
getElementType is deprecated

Reply:
`addr` started life as a StructGEP, so you can use GetElementPtrInst::getTypeAtIndex or directly `jl_array_llvmt-> getElementType(0)` to compute `PPT` without using the deprecated getElementType function

Reply:
Not quite, but I slightly restructured the code in #43721 to avoid getElementType().